core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java

```diff
@@ -343,14 +343,12 @@ public static List<KeyValue> scan(ClientContext context, ScanState scanState, lo
         if (Thread.currentThread().isInterrupted()) {
           throw new AccumuloException("Thread interrupted");
         }
-
-        if ((System.currentTimeMillis() - startTime) / 1000.0 > timeOut) {
+        if (startTime.elapsed(MILLISECONDS) > timeOut.toMillis()) {
```

Comment:
```suggestion
        if (scanTimer.hasElapsed(timeOut)) {
```

Same thing here and the if statement a few lines down. Can be simplified using `hasElapsed()` and potentially be made more clear by renaming the var to `scanTimer` or something like that.